### PR TITLE
Grabpass compatibility improvement with  alpha / antialiasing settings combinations

### PR DESCRIPTION
**Pull request by mvaligursky**
- previously alpha==false case was not handled
- also antialising && !alpha has special handling as Chromium throws errors (even though it works)

fixes https://github.com/playcanvas/engine/issues/2496

### CODE REVIEW

File: src/graphics/device.js [hunk 1]
```diff
@@ -685,7 +685,7 @@ var GraphicsDevice = function (canvas, options) {
     this._spectorCurrentMarker = "";
     // #endif
 
-    this.createGrabPass();
+    this.createGrabPass(options.alpha, options.antialias);
```
Comment (willeastcott) [inline src/graphics/device.js:688]: So, I might be wrong here, but isn't it the case that a WebGL implementation is not required to honor the context creation options? So you could *request* alpha, but not necessarily get it? Same deal with antialising.

Comment (mvaligursky) [reply src/graphics/device.js:688]: That is possible. What other way to detect this is there? These functions are not supported on all platforms:
https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/getFramebufferAttachmentParameter
https://developer.mozilla.org/en-US/docs/Web/API/WebGLRenderingContext/getRenderbufferParameter

Comment (willeastcott) [reply src/graphics/device.js:688]: Hey @kenrussell, do you have any tips for this?

Comment (kenrussell) [reply src/graphics/device.js:688]: Per the spec, https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.2.1 , alpha is required to be honored exactly as specified. antialias is essentially a hint, and might not be honored.

I'm not sure exactly what information you need but you can figure out whether antialiasing was enabled for the back buffer by querying the SAMPLES parameter:

<pre>
let c = document.createElement('canvas');
let gl = c.getContext('webgl');
gl.getParameter(gl.SAMPLES)
> 4

let c2 = document.createElement('canvas');
let gl2 = c2.getContext('webgl', { antialias: false });
gl2.getParameter(gl.SAMPLES)
> 0
</pre>

If there seems to be some bug in Chromium with the combination of antialias:true and alpha:false, please provide a small test case and file a bug on crbug.com . Please include about:gpu from your system when doing so, and ping me the bug ID. Thanks.

Comment (willeastcott) [reply src/graphics/device.js:688]: Thanks for the words of wisdom @kenrussell! Much appreciated. 😄

Comment (mvaligursky) [reply src/graphics/device.js:688]: @kenrussell - thanks for the info, that was helpful.
I have created an issue now - only can reproduce in Chrome on Mac, there was similar issue in the past already, see the description: https://bugs.chromium.org/p/chromium/issues/detail?id=1156756

File: src/graphics/device.js [hunk 2]
```diff
@@ -2123,10 +2148,16 @@ Object.assign(GraphicsDevice.prototype, {
                 texture._parameterFlags = 0;
             }
 
+            // grag framebuffer to be used as a texture - this returns false when not supported for current render pass
```
Comment (slimbuck) [inline src/graphics/device.js:2151]: grag?

Comment (mvaligursky) [reply src/graphics/device.js:2151]: ;) fixing.

**Review by willeastcott — APPROVED** [review]

**Pull request merged**